pipelines/resources/tekton/task/func-deploy/0.1/func-deploy.yaml

```
@@ -24,7 +24,20 @@ spec:
       description: The workspace containing the function project
   steps:
   - name: func-deploy
-    image: "registry.redhat.io/openshift-serverless-1/kn-plugin-func-rhel8:latest"
+    image: "registry.access.redhat.com/ubi8/ubi"
     script: |
+      arch="amd64"
+      machine=$(uname -m)
+      case ${machine} in
+          ppc|ppc64)
+              arch="ppc64le"
+              ;;
+          s390|s390x)
+              arch="s390x"
```

We should fail if machine is not recognized:

```suggestion
          s390|s390x)
              arch="s390x"
              ;;
          *)
              echo "Unsupported archiecture type: ${machine}" 2>&1
              exit 56
```

See: https://linuxize.com/post/bash-redirect-stderr-stdout/

---

just curious here.: why return "56" code, also why should I care about redirect echo standard error here to std out.

---

No particular reason for `56`. It needs to be `!= 0` to indicate failure. It's good to not be `1` as often people use, so you could differentiate errors just by retcode (don't confuse 1:1 relation here).

You should send logs & errors to stderr when scripting. It maybe is not significant in this case, but it's a good habit to have.

---

I know it should be different that 0, just curios about the number choice 56 and yea anyway I think redirect on echo here does not apply. thanks!!